mypy/semanal.py

```diff
@@ -2132,10 +2136,48 @@ def analyze_try_stmt(self, s: TryStmt, visitor: NodeVisitor,
             s.finally_body.accept(visitor)
 
     def visit_with_stmt(self, s: WithStmt) -> None:
+        types = []  # type: List[Type]
+
+        if s.target_type:
+            actual_targets = [t for t in s.target if t is not None]
+            if len(actual_targets) == 0:
+                # We have a type for no targets
+                self.fail('Invalid type comment', s)
+            elif len(actual_targets) == 1:
+                # We have one target and one type
+                types = [s.target_type]
+            elif isinstance(s.target_type, TupleType):
+                # We have multiple targets and multiple types
+                if len(actual_targets) == len(s.target_type.items):
+                    types = s.target_type.items
+                else:
+                    # But it's the wrong number of items
+                    self.fail('Incompatible number of tuple items', s)
+            else:
+                # We have multiple targets and one type
+                self.fail('Tuple type expected for multiple variables', s)
```

Comment:
This error message is also a bit unclear.  (I don't think most users will understand what you mean by "tuple type" here.)  I'd consider something like "Single type given for multiple with targets" or "Multiple with targets require multiple types".